ckanext/hierarchy/plugin.py

```diff
@@ -63,19 +64,67 @@ def get_helpers(self):
                 'group_tree_get_longname': helpers.group_tree_get_longname,
                 'group_tree_highlight': helpers.group_tree_highlight,
                 'get_allowable_parent_groups': helpers.get_allowable_parent_groups,
+                'is_include_children_selected': helpers.is_include_children_selected,
                 }
 
 
-    # IRouter
-    # Redirect organization_read /organization/{id} to custom controller
-    # to include the datasets from the children organizations in the list
-
-    def before_map(self, map):
-        hierarchy_controller = 'ckanext.hierarchy.controller:HierarchyOrganizationController'
-        map.connect('organization_read', '/organization/{id:(?!list$|new$).*}',  controller=hierarchy_controller, action='read', ckan_icon='sitemap')
-        return map
-
-
+    # IPackageController
+    # Modify the search query to include the datasets from
+    # the children organizations in the result list
+    def before_search(self, search_params):
+        ''' If include children selected the query string is modified '''
+        def _children_name_list(children):
+            name_list = []
+            for child in children:
+                name = child.get('name', "")
+                name_list += [name] + _children_name_list(child.get('children', []))
+            return name_list
+
+	query = search_params.get('q', None)
```

Comment:
I looked closed and it appears you have a tab on that line. Please convert it to spaces (because PEP8).